google/resource_dataproc_cluster_test.go

```diff
@@ -0,0 +1,780 @@
+resource "google_dataproc_cluster" "with_service_account" {
+	name   = "dproc-cluster-test-%s"
+	region = "us-central1"
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+
+	service_account = "%s"
+
+	service_account_scopes = [
+        #    The following scopes necessary for the cluster to function properly are
+		#	always added, even if not explicitly specified:
+		#		useraccounts-ro: https://www.googleapis.com/auth/cloud.useraccounts.readonly
+		#		storage-rw:      https://www.googleapis.com/auth/devstorage.read_write
+		#		logging-write:   https://www.googleapis.com/auth/logging.write
+        #
+		#	So user is expected to add these explicitly (in this order) otherwise terraform
+		#   will think there is a change to resource
+		"useraccounts-ro","storage-rw","logging-write",
+
+	    # Additional ones specifically desired by user (Note for now must be in alpha order
+	    # of fully qualified scope name)
+	    "https://www.googleapis.com/auth/monitoring"
+
+	]
+
+}`, rnd, saEmail)
+}
+
+func testAccDataprocCluster_withNetworkRefs(rnd, netName string) string {
+	return fmt.Sprintf(`
+resource "google_compute_network" "dataproc_network" {
+	name = "%s"
+	auto_create_subnetworks = true
+}
+
+resource "google_compute_firewall" "dataproc_network_firewall" {
+	name = "dproc-cluster-test-%s-allow-internal"
+	description = "Firewall rules for dataproc Terraform acceptance testing"
+	network = "${google_compute_network.dataproc_network.name}"
+
+	allow {
+	    protocol = "icmp"
+	}
+
+	allow {
+		protocol = "tcp"
+		ports    = ["0-65535"]
+	}
+
+	allow {
+		protocol = "udp"
+		ports    = ["0-65535"]
+	}
+}
+
+resource "google_dataproc_cluster" "with_net_ref_by_name" {
+	name   = "dproc-cluster-test-%s-name"
+	region = "us-central1"
+	depends_on = ["google_compute_firewall.dataproc_network_firewall"]
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+
+	network = "${google_compute_network.dataproc_network.name}"
+}
+
+resource "google_dataproc_cluster" "with_net_ref_by_url" {
+	name   = "dproc-cluster-test-%s-url"
+	region = "us-central1"
+    depends_on = ["google_compute_firewall.dataproc_network_firewall"]
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+
+	network = "${google_compute_network.dataproc_network.self_link}"
+}
+
+`, netName, rnd, rnd, rnd)
+}
+
+func testAccPreCheckWithServiceAccount(t *testing.T) {
```

Comment:
generally we put all the configs together at the bottom of the test file, so let's move these fns above them